src/nimblepkg/nimscriptapi.nim

```diff
@@ -35,20 +35,26 @@ var
   success = false
   retVal = true
   projectFile = ""
+  outFile = ""
 
 proc requires*(deps: varargs[string]) =
   ## Call this to set the list of requirements of your Nimble
   ## package.
   for d in deps: requiresData.add(d)
 
 proc getParams() =
+  # Called by nimscriptwrapper.nim:execNimscript()
+  #   nim e --flags /full/path/to/file.nims /full/path/to/file.out action
   for i in 2 .. paramCount():
     let
       param = paramStr(i)
-    if param.fileExists():
-      projectFile = param
-    elif param[0] != '-':
-      commandLineParams.add paramStr(i).normalize
+    if param[0] != '-':
+      if projectFile.len == 0:
+        projectFile = param
+      elif outFile.len == 0:
+        outFile = param
+      else:
+        commandLineParams.add param.normalize
```

Comment:
Is this code only parsing arguments that Nimble itself passes to `nim e`? It seems a little fragile to me, and if we control what gets passed to this anyway, can't we just serialise it into something concrete that is easily parsed?

Reply:
`nimscriptwrapper.nim:execNimscript()` calls `nim e` and the params are fixed in order. We don't need to tweak this beyond a point since it is only 3 params and we don't need to add yet another temp file into the mix.